### ISSUE TITLE
Is there a way to remove modules from a MavenLauncher config?

### ISSUE THREAD

**Issue by eoliphan** [INRIA/spoon#4003]
Hi, I'm using Spoon to help with the modernization of some legacy code.  One of of the core projects is a nightmarish maven project, that has 3 subprojects, etc of which have their own subprojects.  The overall project produces multiple war files, etc etc.  In any case, after turning Spoon loose on it, I've discovered that there are some duplicate classes, that don't cause problems for normal builds as they are pulled into artifacts in non-conflicting ways. 

Of course this is causing a problem for Spoon as it attempts to build a model for the entire project.  I've identified modules that I can remove when focusing on the code for one of the generated artifacts, but there doesn't seem to be a way for me to 'edit' the `SpoonPom` i get from calling `.getPomFile` on the launcher.

**Comment by slarse** [comment]
Hi @eoliphan,

You're right that the `SpoonPom` offers no way to remove modules, but there you can provide `MavenLauncher` with a profile filter regex pattern, which only includes modules declared in profiles that match the regex. If it's possible to alter the core pom file that declares the modules to do so in different profiles (as appropriate to avoid duplication), that might be an option.

Another option is to set `Environment.setIgnoreDuplicateDeclarations(true)`, but that may cause other issues as the built model is necessarily incorrect in places.

The final option is to just use the standard launcher and provide it with any necessary classpath to resolve dependencies from, and then selectively add input sources to it.

**Comment by eoliphan** [comment]
Yeah, i got around it temporarily with ignoring duplicate defs, but to your point it's probably not going to work for some of the more sophisticated stuff i need to do.  I can't really mess with the pom files themselves and given the complexity, i'd like to avoid adding stuff manually if i can.  I'm going to give the regex filter a whirl.  Thanks!
